Library/Homebrew/utils/shell.rb

```diff
@@ -0,0 +1,71 @@
+module Utils
+  SHELL_PROFILE_MAP = {
+    :bash => "~/.bash_profile",
+    :csh => "~/.cshrc",
+    :fish => "~/.config/fish/config.fish",
+    :ksh => "~/.kshrc",
+    :sh => "~/.bash_profile",
+    :tcsh => "~/.tcshrc",
+    :zsh => "~/.zshrc",
+  }.freeze
+
+  module Shell
+    # take a path and heuristically convert it
+    # to a shell, return nil if there's no match
+    def self.path_to_shell(path)
+      # remove leading - from login shell
+      shell_name = File.basename(path).strip_prefix("-")
```

Comment:
as mentioned in https://github.com/Homebrew/brew/pull/201/files#r62985414, `strip_prefix("-")` can be avoid if use different ps command argument.